akka-stream/src/main/scala/akka/stream/impl/QueueSource.scala

```diff
@@ -40,7 +40,7 @@ import scala.util.control.NonFatal
   override def createLogicAndMaterializedValue(inheritedAttributes: Attributes) = {
     val completion = Promise[Done]
 
-    val stageLogic = new GraphStageLogic(shape) with OutHandler with SourceQueueWithComplete[T] {
+    val stageLogic = new GraphStageLogic(shape) with OutHandler with SourceQueueWithComplete[T] with StageLogging {
```

Comment:
better logger name if you override `logSource` in `StageLogging`
```
override protected def logSource = classOf[QueueSource]
```

in all places